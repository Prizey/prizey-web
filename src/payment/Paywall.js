import React from 'react'
import { List } from 'croods'
import get from 'lodash/get'
import { withStyles } from '@material-ui/core/styles'
import { Typography, Button } from '@material-ui/core'
import Layout from 'design/Layout/Layout'
import { Redirect, Link } from '@reach/router'

import GoBack from 'design/GoBack/GoBack'
import UserBalance from 'design/UserBalance'

import getButtonText from './purchaseDescription'

const styles = theme => ({
  button0: {
    '&:hover': {
      background: get(theme.palette, 'paywall.first'),
    },
    background: get(theme.palette, 'paywall.first'),
  },
  button1: {
    '&:hover': {
      background: get(theme.palette, 'paywall.second'),
    },
    background: get(theme.palette, 'paywall.second'),
  },
  buttonGroup: {
    marginBottom: theme.spacing.md,
    marginTop: theme.spacing.md,
  },
  icon: {
    height: theme.spacing.md,
    marginRight: theme.spacing.xs,
  },
})

const PaywallScreen = withStyles(styles)(
  ({ classes, buttons, buyMore = false }) => (
    <Layout leftIcon={buyMore ? <UserBalance /> : <GoBack to="/" />}>
      <Typography align="center" variant="h5">
        {buyMore
          ? "You don't have enough for that game, add more to your account to play."
          : 'Purchase diamonds to play, win something every time.'}
      </Typography>

      <div className={classes.buttonGroup}>
        {buttons.map(({ id, ticketAmount, price }, idx) => (
          <Button
            key={idx}
            variant="contained"
            color="primary"
            fullWidth
            className={classes[`button${idx}`]}
            component={props => <Link to={`/payment/${id}`} {...props} />}
          >
            <img
              src="/icons/diamond.png"
              alt="diamond"
              className={classes.icon}
            />
            {getButtonText(ticketAmount, price)}
          </Button>
        ))}
      </div>

      <Typography align="center" variant="body2" color="textSecondary">
        By proceeding, you agree you&apos;re 18+.
      </Typography>
    </Layout>
  ),
)

export default ({ currentUser, buyMore = false }) =>
  currentUser ? (
    <List
      name="purchaseOptions"
      path="/purchase_options"
      render={buttons => <PaywallScreen buyMore={buyMore} buttons={buttons} />}
    />
  ) : (
    <Redirect
      to={`/sign-in?next=/${buyMore ? 'buy-more' : 'buy-diamonds'}`}
      noThrow
    />
  )
