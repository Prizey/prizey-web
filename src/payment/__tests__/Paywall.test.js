import React from 'react'
import renderer from 'react-test-renderer'
import Paywall from '../Paywall'

const mockList = [
  { id: 1, price: '10.0', ticketAmount: 10 },
  { id: 2, price: '25.0', ticketAmount: 25 },
  { id: 3, price: '50.0', ticketAmount: 50 },
]

jest.mock('../../design/Layout/RegisterPageView')

jest.mock('croods', () => ({
  List: props => (
    <div {...props}>
      {props.children} - Render: {props.render(mockList)}
    </div>
  ),
}))

it('renders correctly', () => {
  const tree = renderer.create(<Paywall currentUser />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('redirects correctly', () => {
  const tree = renderer.create(<Paywall />).toJSON()
  expect(tree).toMatchSnapshot()
})
