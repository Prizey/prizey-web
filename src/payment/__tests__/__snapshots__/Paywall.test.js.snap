// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirects correctly 1`] = `null`;

exports[`renders correctly 1`] = `
<div
  name="purchaseOptions"
  path="/purchase_options"
  render={[Function]}
>
   - Render: 
  <div
    className="Component-root-12"
  >
    <div
      className="Component-header-10"
    >
      <a
        aria-current="page"
        aria-label="Go Back"
        className="MuiButtonBase-root-20 MuiIconButton-root-14 Component-root-13"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiIconButton-label-19"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-23 MuiSvgIcon-fontSizeLarge-31"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
            />
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-94"
        />
      </a>
    </div>
    <div
      className="Component-container-7"
    >
      <img
        alt="Prizey"
        className="Component-logoImage-11"
        src="/logo.png"
      />
      <div
        className="Component-caption-5"
      />
      <div
        className="Component-children-6"
      >
        <h5
          className="MuiTypography-root-32 MuiTypography-h5-48 MuiTypography-alignCenter-55"
        >
          Purchase diamonds to play, win something every time.
        </h5>
        <div
          className="Component-buttonGroup-3"
        >
          <a
            className="MuiButtonBase-root-20 MuiButton-root-68 MuiButton-contained-79 MuiButton-containedPrimary-80 MuiButton-raised-82 MuiButton-raisedPrimary-83 MuiButton-fullWidth-93 Component-button0-1"
            href="/payment/1"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label-69"
            >
              <img
                alt="diamond"
                className="Component-icon-4"
                src="/icons/diamond.png"
              />
              10 Diamonds = $ 10.00
            </span>
            <span
              className="MuiTouchRipple-root-94"
            />
          </a>
          <a
            className="MuiButtonBase-root-20 MuiButton-root-68 MuiButton-contained-79 MuiButton-containedPrimary-80 MuiButton-raised-82 MuiButton-raisedPrimary-83 MuiButton-fullWidth-93 Component-button1-2"
            href="/payment/2"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label-69"
            >
              <img
                alt="diamond"
                className="Component-icon-4"
                src="/icons/diamond.png"
              />
              25 Diamonds = $ 25.00
            </span>
            <span
              className="MuiTouchRipple-root-94"
            />
          </a>
          <a
            className="MuiButtonBase-root-20 MuiButton-root-68 MuiButton-contained-79 MuiButton-containedPrimary-80 MuiButton-raised-82 MuiButton-raisedPrimary-83 MuiButton-fullWidth-93"
            href="/payment/3"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label-69"
            >
              <img
                alt="diamond"
                className="Component-icon-4"
                src="/icons/diamond.png"
              />
              50 Diamonds = $ 50.00
            </span>
            <span
              className="MuiTouchRipple-root-94"
            />
          </a>
        </div>
        <p
          className="MuiTypography-root-32 MuiTypography-body2-40 MuiTypography-colorTextSecondary-65 MuiTypography-alignCenter-55"
        >
          By proceeding, you agree you're 18+.
        </p>
      </div>
      <div
        className="Component-footer-8"
      >
        <a
          className="Component-footerLink-9"
          href="https://termly.io/en/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            className="MuiTypography-root-32 MuiTypography-caption-42 MuiTypography-colorTextSecondary-65 MuiTypography-alignCenter-55"
          >
            TERMS
          </span>
        </a>
        <a
          className="Component-footerLink-9"
          href="https://termly.io/en/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            className="MuiTypography-root-32 MuiTypography-caption-42 MuiTypography-colorTextSecondary-65 MuiTypography-alignCenter-55"
          >
            PRIVACY
          </span>
        </a>
        <a
          className="Component-footerLink-9"
          href="https://termly.io/en/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            className="MuiTypography-root-32 MuiTypography-caption-42 MuiTypography-colorTextSecondary-65 MuiTypography-alignCenter-55"
          >
            FAIRNESS
          </span>
        </a>
        <a
          className="Component-footerLink-9"
          href="https://termly.io/en/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            className="MuiTypography-root-32 MuiTypography-caption-42 MuiTypography-colorTextSecondary-65 MuiTypography-alignCenter-55"
          >
            ALL PRIZES
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
